### ISSUE TITLE
Visual Studio uses the default client name for the UserAgent string

### ISSUE THREAD

**Issue by emgarten** [NuGet/Home#5853]
Visual Studio uses the default user agent string of `NuGet Client V3`. It should instead use a meaningful name so that it is clear which client was used on nuget.org stats. It would also help avoid conflicts with 3rd party tools consuming the nuget libraries directly which could also lead to downloads under this name.

```
GET https://api.nuget.org/v3-flatcontainer/newtonsoft.json/6.0.4/newtonsoft.json.6.0.4.nupkg HTTP/1.1
user-agent: NuGet Client V3/4.3.1 (Microsoft Windows NT 10.0.15063.0, VS Enterprise/15.0)
X-NuGet-Client-Version: 4.3.1
Accept-Language: en-US
Host: api.nuget.org
Accept-Encoding: gzip, deflate
```

Based on https://www.nuget.org/stats/packages/Newtonsoft.Json?groupby=ClientName&groupby=ClientVersion it looks like All 3.x and 4.x versions of NuGet have this issue.

**Comment by emgarten** [comment]
@anangaur do you have a suggestion for a useful name here?

**Comment by emgarten** [comment]
Moving to pri 0 based on the discussion yesterday. We still need a name for this.

//cc @rrelyea

**Comment by anangaur** [comment]
@emgarten Can you list all the strings we currently use and the clients that send them? Something like:

| Client | Current UserAgent str | Proposed UserAgent str |
|:----|:---|:---|
| nuget.exe | ? | @anangaur - tbd|

**Comment by emgarten** [comment]
You can see them on https://www.nuget.org/stats/packages/Newtonsoft.Json?groupby=ClientName&groupby=ClientVersion

Look for all 4.4 versions

**Comment by anangaur** [comment]
Is **NuGet MSBuild Task** not coming from VS? Is it from `msbuild.exe /t:restore` commandline?

**Comment by anangaur** [comment]
If it is about what VS should send, then I think the string should have:
* A customer identifiable string to mean VS
* Version number should be able to identify the VS/NuGet version including previews

So what about:
user-agent: **NuGet VS VSIX/4.3.1**

I am not sure why V3 is important in the existing UserAgent string especially when the version is present in the user agent?

**Comment by emgarten** [comment]
MSBuild Task (core and desktop) only come from command line, not from VS or nuget.exe. 

V3 isn't important, that string is the default set by the library and should be overridden by the actual app using it.

I like `NuGet VS VSIX`, I believe the version number is auto appended to everything as part of the formatting, and nuget.org understands that format also.

**Comment by nkolev92** [comment]
Overall string would look something like:

NuGet VS VSIX/4.7.0 (Microsoft Windows NT 10.0.14393.0, VS Enterprise/15.0)

**Comment by jainaashish** [comment]
Merged into release-4.6.0-rtm 336c84d0cded455ab38965e01347cfba049cfd9d
